test/plugin/test_out_google_cloud.rb

```diff
@@ -191,6 +191,19 @@ def test_parse_severity
     assert_equal('DEFAULT', test_obj.parse_severity('er'))
   end
 
+  def test_tags
+    setup_gce_metadata_stubs
+    [123, 'test', 'germanß', 'chinese中', 'specialCharacter_-.'].each do |tag|
+      setup_logging_stubs([COMPUTE_PARAMS.merge(log_name: tag.to_s)]) do
+        @logs_sent = []
+        d = create_driver(APPLICATION_DEFAULT_CONFIG, tag)
+        d.emit('msg' => log_entry(0))
+        d.run
+      end
+      verify_log_entries(1, COMPUTE_PARAMS, 'structPayload')
```

Comment:
For non-grpc, log name is reflected in the url instead of in the log entry itself. I just added some comments to clarify this as well:

`  # For non-grpc path, log name is reflected in the url, so we are stubbing the
  # corresponding url containing the converted tag. To verify the tag is
  # converted properly, we simply check the log entry is recorded successfully,
  # and no exception (e.g. stubbing missing for certain url) is thrown.`

The majority of the test cases are identical between grpc and non-grpc, but the setup_logging_stub part requires some special treatment for non-grpc. Also the way to verify converted tag is slightly different. Thus this ended up in two separate files instead of in the `base_test` (which i find pretty annoying).